### PR TITLE
Update starter files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add version to output
- Update README
- Rename test directory to tests
- Update ignored files
- Update shell scripts
- Simplify TODOs
- Add sanitize option

### CODE REVIEW

**Review by iHiD — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -53,10 +46,23 @@ To run the tests to verify the behavior of the test runner, do the following:
 1. Open a terminal in the project's root
 2. Run `./bin/run-tests.sh`
 
-These are [golden tests][golden] that compare the `results.json` generated by running the current state of the code against the "known good" `test/<test-name>/results.json`. All files created during the test run itself are discarded.
+These are [golden tests][golden] that compare the `results.json` generated by running the current state of the code against the "known good" `tests/<test-name>/results.json`. All files created during the test run itself are discarded.
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply README.md:49]: You might want to write `golden tests (snapshot tests)`, which is a common term used in a lot of languages / test libs / frameworks.

File: bin/run-in-docker.sh [hunk 2]
```diff
@@ -2,41 +2,42 @@
...
-# Builds the Dockerfile
-# Runs the docker image passing along the initial arguments
+# Run the test runner on a solution using the test runner Docker image.
+# The test runner Docker image is built automatically.
 
 # Arguments:
 # $1: exercise slug
-# $2: **RELATIVE** path to solution folder (with trailing slash)
-# $3: **RELATIVE** path to output directory (with trailing slash)
+# $2: absolute path to solution folder
```
Comment (SleeplessByte) [inline bin/run-in-docker.sh:10]: The trailing slash part is probably still important. `absolute` only means it starts with a drive, mount or `/`

File: bin/run-tests.sh [hunk 3]
```diff
@@ -1,11 +1,43 @@
...
-    if [ "$testname" != output ] && [ -f "${testdir}/results.json" ]; then
-        bin/run.sh "$testname" "$testdir" "$output_dir"
+# Output:
+# Outputs the diff of the expected test results against the actual test results
+# generated by the test runner.
+
+# Example:
+# ./bin/run-tests.sh
+
+exit_code=0
```
Comment (SleeplessByte) [inline bin/run-tests.sh:14]:
```suggestion
set -euo pipefail

exit_code=0
```

File: bin/run.sh [hunk 4]
```diff
@@ -1,26 +1,62 @@
...
-cp "${INPUT_DIR}/results.json" "$OUTPUT_DIR"
+# If any required arguments is missing, print the usage and exit
+if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
+    echo "usage: ./bin/run.sh exercise-slug /absolute/path/to/two-fer/solution/folder/ /absolute/path/to/output/directory/"
+    exit 1
+fi
 
-echo "$SLUG: comparing ${OUTPUT_DIR}/results"
-diff "${INPUT_DIR}/results.json" "${OUTPUT_DIR}/results.json"
+slug="$1"
```
Comment (SleeplessByte) [inline bin/run.sh:24]:
```suggestion
set -euo pipefail

slug="$1"
```

File: bin/run.sh [hunk 5]
```diff
@@ -1,26 +1,62 @@
...
-INPUT_DIR="$2"
-OUTPUT_DIR="$3"
+# Synopsis:
+# Run the test runner on a solution.
 
-echo "$SLUG: testing..."
-# PLACEHOLDER - Insert call to run your language tests here
+# Arguments:
+# $1: exercise slug
+# $2: absolute path to solution folder
```
Comment (SleeplessByte) [inline bin/run.sh:8]: The trailing slash part is probably still important. `absolute` only means it starts with a drive, mount or `/`

**Review by SleeplessByte — COMMENTED** [review]
Great stuff.

**Pull request merged**